sqsworkers/crew.py

```diff
@@ -157,25 +158,41 @@ def poll_queue(self):
                 MessageAttributeNames=['All'],
                 MaxNumberOfMessages=self.max_number_of_messages,
                 WaitTimeSeconds=self.wait_time)
-
             if len(messages) > 0:
-                for message in messages:
-                    self.logger.info('processing %s messages %s' % (len(messages), messages))
-                    processor = self.crew.MessageProcessor(message)
-                    self.crew.statsd.increment('process.record.start', 1, tags=[])
-                    processed = processor.start()
-                    if processed:
-                        deleted = self.queue.delete_messages(
-                            Entries=[{
-                                'Id': message.message_id,
-                                'ReceiptHandle': message.receipt_handle
-                            }]
-                        )
-                        self.crew.statsd.increment('process.record.success', 1, tags=[])
-                        self.logger.info('%s messages processed successfully and deleted %s' % (len(messages), deleted))
-                    else:
-                        self.crew.statsd.increment('process.record.failure', 1, tags=[])
-
+                try:
+                    for message in messages:
```

Comment:
Can we move the try inside the loop so that the try/except is per message as opposed to the whole set of messages, if that makes sense?

Reply:
yes, it does. Will make that change.